### PR TITLE
Fix ESP32-C3 using v2.0.0-alpha1 support

### PR DESCRIPTION
**Pull request by agners**
# What does this implement/fix? 

Quick description and explanation of changes

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [x] ESP32-C3
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
esphome:
  name: esp-c3
  platform: ESP32
  board: nodemcu-32s
  platformio_options:
    board_build.mcu: esp32c3
    platform: https://github.com/Jason2866/platform-espressif32.git#feature/arduino-c3
    platform_packages: framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git#2.0.0-alpha1
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/api/user_services.cpp [hunk 1]
```diff
@@ -15,7 +15,7 @@ template<> std::string get_execute_arg_value<std::string>(const ExecuteServiceAr
 template<> std::vector<bool> get_execute_arg_value<std::vector<bool>>(const ExecuteServiceArgument &arg) {
   return arg.bool_array;
 }
-template<> std::vector<int> get_execute_arg_value<std::vector<int>>(const ExecuteServiceArgument &arg) {
+template<> std::vector<int32_t> get_execute_arg_value<std::vector<int32_t>>(const ExecuteServiceArgument &arg) {
```
Comment (OttoWinter) [inline esphome/components/api/user_services.cpp:18]: What is `sizeof(int)` on ESP32-C3? Is it 8 bytes? Maybe that can cause some issues in other parts of the codebase

**Review by OttoWinter — APPROVED** [review]

File: esphome/components/api/user_services.cpp (hunk 1 above)
Comment (agners) [reply esphome/components/api/user_services.cpp:18]: Afaik, its a 32-bit machine, so I would expect 4 bytes.

I was wondering why this happens, my guess currently is that new gcc/libc (or the RISC-V implementation thereof) added some provisioning to guard against implicitly interchanging the two types. You can find the exact error message in the commit:
https://github.com/esphome/esphome/pull/2035/commits/0180c01c740b14dddd9cd80fb9a717f94eaad49b

Comment (agners) [reply esphome/components/api/user_services.cpp:18]: It seems that int32_t is defined as `long int` for 32-bit RISC-V, which is considered a different type. See also:
https://godbolt.org/z/KnWT1xj3o

Comment (OttoWinter) [reply esphome/components/api/user_services.cpp:18]: Interesting, good to know

**Comment by markusg1234** [comment]
Hi,
Thankyou for your efforts. I now have a ESP32-C3 now online with ESPHome but unable to use the ADC inputs as it expects the ADC to be between GPIO 32-39 with using board as nodemcu-32s.
Is there a workaround to map the C3 pins to allow the ADC for GPIO pins 0-5.
Regards,
Markus.

**Comment by digiblur** [comment]
Looks like there's some work to be done on the pin mapping and other functions.

/root/.platformio/lib/ESPAsyncWebServer-esphome/src/AsyncWebSocket.cpp: In member function 'IPAddress AsyncWebSocketClient::remoteIP()':
/root/.platformio/lib/ESPAsyncWebServer-esphome/src/AsyncWebSocket.cpp:840:28: error: call of overloaded 'IPAddress(unsigned int)' is ambiguous
         return IPAddress(0U);
                            ^

**Comment by agners** [comment]
@markusg1234 proper board support is still missing indeed. Working on it.

@digiblur this issue has already been identified and fixes proposed, see https://github.com/esphome/feature-requests/issues/1217

Let's move further discussion related to ESP32-C3 to https://github.com/esphome/feature-requests/issues/1217 since this PR is merged.

**Comment by digiblur** [comment]
> @markusg1234 proper board support is still missing indeed. Working on it.
> 
> @digiblur this issue has already been identified and fixes proposed, see [esphome/feature-requests#1217](https://github.com/esphome/feature-requests/issues/1217)
> 
> Let's move further discussion related to ESP32-C3 to [esphome/feature-requests#1217](https://github.com/esphome/feature-requests/issues/1217) since this PR is merged.

Thanks for the links...getting confusing with the same names of threads.

**Pull request merged**